semgrep/semgrep/rule_lang.py

```diff
@@ -391,24 +391,24 @@ def _validation_error_message(error: jsonschema.exceptions.ValidationError) -> s
     any_of_invalid_keys = set()
     required = set()
     banned = set()
-    for m in (c.message for c in contexts):
-        if RuleValidation.BAD_TYPE_SENTINEL in m:
-            bad_type.add(m)
-        if RuleValidation.INVALID_SENTINEL in m:
-            ix = m.find(RuleValidation.INVALID_SENTINEL)
+    for context in contexts:
```

Comment:
We can remove YAML string parsing here altogether, and instead rely on the member variables `validator_value` and `instance` from here: https://python-jsonschema.readthedocs.io/en/stable/errors/#jsonschema.exceptions.ValidationError. The tests were failing because this code was relying on implementation details of `ruamel`. This change fixes up the tests too :+1: